### ISSUE TITLE
UP028 for multiple loops does not work

### ISSUE THREAD

**Issue by spaceby** [astral-sh/ruff#13266]
UP028 does not work when there are multiple yield loops in a function

Code example:
```python
def text_not_ok():
    a = [1, 2, 3]
    for photo in a:
        yield photo
    for photo in a:
        yield photo
```

https://play.ruff.rs/b9748aaf-8e8c-45cd-8138-20f41013b979

**Comment by AlexWaygood** [comment]
I think this is because of this logic here: https://github.com/astral-sh/ruff/blob/c3bcd5c8422b79bd5cbc874a98f70cdb92e8d445/crates/ruff_linter/src/rules/pyupgrade/rules/yield_in_for_loop.rs#L98-L112

I.e., Ruff believes that some of the variables bound by the first loop might be referenced in the second loop, so it comes to the conclusion that it won't necessarily be safe to rewrite them as `yield from` statements. If you change the second loop to use different variable names, it flags both loops with `UP028`: https://play.ruff.rs/50ee023e-60f4-4a77-a870-3378a6c68b72

It would be possible to improve Ruff's behaviour here, I think, but it might add some complexity to our implementation

**Comment by spaceby** [comment]
Then you need to improve ruff's behavior
Or add another rule that tracks repeating loops
Or add a warning about this behavior to the documentation
